### PR TITLE
Added MAYK to the food menu.

### PR DESCRIPTION
**Pull request by zeriaxdev**
I added MAYK (Maunulan yhteiskoulu) to the list, as it is my current school.
I'd really love to see it in production! Thank you!

Changes:
```
        modified:   README.md
        new file:   src/handlers/mayk.ts
        modified:   src/main.ts
        new file:   src/parsers/mayk.ts
```

### CODE REVIEW

**Comment by developerfromjokela** [comment]
Thanks for the PR!

I'll have a look tomorrow, and possibly deploy to production (if it makes through to merging) this week.

**Comment by zeriaxdev** [comment]
Thanks! I hope nothing breaks 😂

**Comment by developerfromjokela** [comment]
Seems like it doesn't work correctly? Currently it shows empty response, despite the mayk website containing the food menu.

image

image

File: src/parsers/mayk.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    let contentBox = document.querySelector("#ruoka-viikko-wrapper");
+    if (contentBox !== null) {
+        let items: Day[] = [];
+        let children = contentBox.querySelectorAll('.ruoka-template-header');
+        let currentDayDate: undefined|Moment = undefined;
+        children.forEach(child => {
+            if (child.querySelector('.ruoka-header-pvm') && !currentDayDate) {
+                let regexResult = dateRegex.exec(child.textContent);
+                if (regexResult != null)
+                    currentDayDate = moment(regexResult[0], "DD.MM.YYYY").startOf('day');
```
Comment (developerfromjokela) [inline src/parsers/mayk.ts:28]: Food menu supplies a week name (maanantai,tiistai,etc), in `.ruoka-header-pvm`.
The way to get real date is to first get `ruokalista-viikko` value, in order to get a date in return.
With that date, just change the week day according to `.ruoka-header-pvm`.

But these food menus are often unpredictable, most of them written by hand, thus they sometimes change unexpectedly.

**Review by developerfromjokela — CHANGES_REQUESTED** [review]
Parser needs enhancements for date handling.

File: src/parsers/mayk.ts (hunk 1 above)
Comment (zeriaxdev) [reply src/parsers/mayk.ts:28]: I'm truly sorry for these issues...

> But these food menus are often unpredictable, most of them written by hand, thus they sometimes change unexpectedly.

Sadly, it's true. Sometimes dates are just missing.

Comment (developerfromjokela) [reply src/parsers/mayk.ts:28]: If you could implement a simple check, if it contains the week instead, and take appropriate action?

**Comment by developerfromjokela** [comment]
@zeriaxdev should I close this PR or you're still working on this?

**Comment by zeriaxdev** [comment]
> @zeriaxdev should I close this PR or you're still working on this?

I think you should. I'm not currently working on this, and the website itself has some issues.

**Pull request closed without merging**